gaphor/ui/errorhandler.py

```diff
@@ -10,31 +10,48 @@
 
 from gi.repository import Gtk
 
+if Gtk.get_major_version() == 4:
+    from gi.repository import Adw
+
 from gaphor.i18n import gettext
 
 
 def error_handler(message, secondary_message="", window=None):
     exc_type, exc_value, exc_traceback = sys.exc_info()
 
-    dialog = Gtk.MessageDialog(message_type=Gtk.MessageType.ERROR, text=message)
-    dialog.set_transient_for(window)
-
-    if __debug__ and exc_traceback and sys.stdin.isatty():
-        dialog.props.secondary_text = (
-            f"{secondary_message}\n\n" if secondary_message else ""
-        ) + gettext(
-            "It looks like Gaphor is started from the command line. Do you want to open a debug session?"
-        )
-        dialog.add_buttons(gettext("Close"), 0, gettext("Start debug session"), 100)
+    debug_message = (f"{secondary_message}\n\n" if secondary_message else "") + gettext(
+        "It looks like Gaphor is started from the command line. Do you want to open a debug session?"
+    )
+    if Gtk.get_major_version() == 3:
+        dialog = Gtk.MessageDialog(message_type=Gtk.MessageType.ERROR, text=message)
+        dialog.set_transient_for(window)
+
+        if __debug__ and exc_traceback and sys.stdin.isatty():
+            dialog.props.secondary_text = debug_message
+            dialog.add_buttons(gettext("Close"), 0, gettext("Start Debug Session"), 100)
+        else:
+            dialog.props.secondary_text = secondary_message
+            dialog.add_button(gettext("Close"), 0)
+        dialog.set_modal(True)
     else:
-        dialog.props.secondary_text = secondary_message
-        dialog.add_button(gettext("Close"), 0)
+        dialog = Adw.MessageDialog.new(
+            window,
+            message,
+        )
+        dialog.set_body(debug_message)
+        if __debug__ and exc_traceback and sys.stdin.isatty():
```

Comment:
It looks like the debug message is not shown here

Reply:
I renamed the variable to `body`, it wasn't a good choice of names. Is this more clear now?